### PR TITLE
Some fixes for the Ubuntu/Debian and Fedora packaging

### PR DESCRIPTION
**Pull request by dreibh**
Some fixes for the Ubuntu/Debian and Fedora packaging. The packaging processes broke after merging recent pull requests.

### CODE REVIEW

**Comment by weinrank** [comment]
We agreed in Cork to put the socket-api stuff in a separate repository, this PR does the opposite...

**Comment by dreibh** [comment]
The pull request just fixes the package build issues. It is not intended to separate the two libraries. This would be a different task.

**Comment by weinrank** [comment]
It re-enables building the socket-api by default which has been disabled by intention.

**Comment by dreibh** [comment]
No. In CMakeLists.txt (version 40774dc17dbf189b0a8beaebb655457e2226d484 of June 27):
OPTION(SOCKET_API "Include the socket API" 0)

That is, building the sockets API is *off* by default (as it was before).

I just verified the build behaviour:
# git clean -df ; cmake . && make -j12 && ls -l socketapi/*.so*
...
ls: klarte ikke å åpne 'socketapi/*.so*': Fila eller mappa finnes ikke

The sockets API is only built when explicitly requested with SOCKET_API=1:
# git clean -df ; cmake -DSOCKET_API=1 . && make -j12 && ls -l socketapi/*.so*
...
lrwxrwxrwx 1 dreibh dreibh     22 juni  30 10:26 socketapi/libneat-socketapi.so -> libneat-socketapi.so.0
lrwxrwxrwx 1 dreibh dreibh     32 juni  30 10:26 socketapi/libneat-socketapi.so.0 -> libneat-socketapi.so.0.0.1~td158
-rwxrwxr-x 1 dreibh dreibh 179264 juni  30 10:26 socketapi/libneat-socketapi.so.0.0.1~td158

**Comment by weinrank** [comment]
https://github.com/NEAT-project/neat/blob/98c49addd07f1886b2de0d49694f75b305b0d985/autogen.sh#L4

https://github.com/NEAT-project/neat/blob/98c49addd07f1886b2de0d49694f75b305b0d985/debian/rules#L10

https://github.com/NEAT-project/neat/blob/98c49addd07f1886b2de0d49694f75b305b0d985/rpm/libneat.spec#L134

**Comment by dreibh** [comment]
debian/rules and rpm/libneat.spec need a full build including the sockets API, of course. The resulting binary packages for Debian/Ubuntu and Fedora can of course be installed separately (i.e. pbuilder/mock will build separate packages for core library, sockets API, development, etc.). However, the rules in these files are only relevant for Debian/Ubuntu and Fedora packaging. Manually building from source (with "cmake") is not affected by these files.

This pull requests only fixes the packaging (which currently does not work in "master"), it does not separate both API libraries. This is a different task.

File: debian/rules [hunk 1]
```diff
@@ -7,7 +7,7 @@ export DEB_BUILD_MAINT_OPTIONS = hardening=+all
 	# --with=sphinxdoc
 
 override_dh_auto_configure:
-	dh_auto_configure -- -DUSRSCTP_SUPPORT=1 -DSCTP_MULTISTREAMING=1 -DFLOW_GROUPS=1
+	dh_auto_configure -- -DSOCKET_API=1 -DUSRSCTP_SUPPORT=1 -DSCTP_MULTISTREAMING=1 -DFLOW_GROUPS=1
```
Comment (weinrank) [inline debian/rules]: -DSOCKET_API=1  ?

File: rpm/libneat.spec [hunk 2]
```diff
@@ -131,7 +131,7 @@ Requires: %{name}-socketapi = %{version}-%{release}
 %setup -q
 
 %build
-%cmake -DCMAKE_INSTALL_PREFIX=/usr -DUSRSCTP_SUPPORT=0 -DSCTP_MULTISTREAMING=1 -DFLOW_GROUPS=1 .
+%cmake -DCMAKE_INSTALL_PREFIX=/usr -DSOCKET_API=1 -DUSRSCTP_SUPPORT=0 -DSCTP_MULTISTREAMING=1 -DFLOW_GROUPS=1 .
```
Comment (weinrank) [inline rpm/libneat.spec]: -DSOCKET_API=1 ?

**Comment by dreibh** [comment]
Again: debian/rules and rpm/libneat.spec are the configuration files for the Debian and Fedora packaging. They need this option for a full build from source. The resulting binary packages are of course separated into callback API library, sockets API library, etc..

**Pull request merged**